src/main/cpp/Monopoly.cpp

```diff
@@ -240,13 +280,243 @@ Monopoly::print(string s, bool wait, bool reset) {
+			print("You moved to " + newSpot->getName() + "!");			
+			landOnRailroadTile(2);
+			break;
+		case 4:
+			print("Bank pays you dividend of $50.", true, false);
+			player->addMoney(50);
+			break;
+		case 5:
+			print("Get out of Jail Free.", true, false);
+			player->setOutOfJailCard(true);
+			break;
+		case 6:
+			print("Go back 3 spaces.", true, false);
+			movePlayerToSpot((originalSpot + 37) % 40, false);
+			break;
+		case 7:
+			print("Go to Jail--go directly to jail--do not pass Go, do not collect $200.", true, false);
+			sendPlayerToJail(player);
+			break;
+		case 8:
+			print("Make general repairs on all your property--for each house pay $25--for each hotel $100.", true, false);
+			payHouseRepairs(player);
+			break;
+		case 9:
+			print("Pay poor tax of $15.", true, false);
+			player->loseMoney(15);
+			break;
+		case 10:
+			print("Take a trip to Reading Railroad--if you pass Go, collect $200.", true, false);
+			movePlayerToSpot(5);
+			landOnRailroadTile();
+			break;
+		case 11:
+			print("Take a walk on the Boardwalk--advance token to Boardwalk.", true, false);
+			movePlayerToSpot(39);
+			landOnHouseTile();
+			break;
+		case 12:
+			print("You have been elected Chairman of the Board. Pay each player $50.", true, false);
+			payEachPlayer(50);
+			break;
+		case 13:
+			print("Your building and loan matures--collect $150.", true, false);
+			player->addMoney(150);
+			break;
+		case 14:
+			print("You have won a crossword competition--collect $100.", true, false);
+			player->addMoney(100);
+			break;
+		case 15:
+			print("Advance to St. Charles Place--if you pass Go, collect $200.", true, false);
+			movePlayerToSpot(11);
+			landOnHouseTile();
+			break;
+	}
+}
+
+void
+Monopoly::sendPlayerToJail(shared_ptr<Player> player) {
+	if (player->hasOutOfJailCard()) {
+		print("You have a 'Get Out of Jail' card! You skipped this time!");
+		player->setOutOfJailCard(false);
+		return;
+	}
+	player->goToJail();
+}
+
+void
+Monopoly::moveCurrentPlayer(int roll) {
+	shared_ptr<Player> player = getCurrentPlayer();
+	int origionalSpot = player->getLocationNum();
+	player->move(roll);
+
+	if(player->getLocationNum() < origionalSpot) {
+		print("You recieved $200!");
+		player->addMoney(200);
+	}
+}
+
+void
+Monopoly::movePlayerToSpot(int val, bool getBonus) {
+	shared_ptr<Player> player = getCurrentPlayer();
+	int origionalSpot = player->getLocationNum();
+	player->goToSpot(val);
+
+	if(player->getLocationNum() < origionalSpot && getBonus) {
+		print("You recieved $200!");
+		player->addMoney(200);
+	}
+}
+
+void
+Monopoly::payEachPlayer(int val) {
+	shared_ptr<Player> currentPlayer = getCurrentPlayer();
+	line->nextTurn();
+	for(int i = 0; i < line->getNumPlayers() - 1; i++) {
+		getCurrentPlayer()->addMoney(50);
+		line->nextTurn();
+	}
+	currentPlayer->loseMoney((line->getNumPlayers() - 1) * 50);
+
+}
+
+void
+Monopoly::payHouseRepairs(shared_ptr<Player> player) {
+	list<HouseTile *> ownedHouses = getOwnedHouses(player);
```

Comment:
Suggest shared_ptr's here

Reply:
I don't think I'll be able to do this because of the type casting issue.